### ISSUE TITLE
Allow for the CursorIcon to be configured on startup

### ISSUE THREAD

**Issue by Jay-Madden** [neovide/neovide#2644]
**Is your feature request related to a problem? Please describe.**

I am used to neovim running in a terminal that makes the cursoricon shape a text bar. Neovide uses the default arrow pointer when focused and I have no been able to get used to it.

**Describe the solution you'd like**

Winit offers the `CursorIcon` enum with the accompanying `window.set_cursor_icon` method. I propose that when the window is initialized the config values are read and parsed into that enum to set the cursor icon to a configured value.

List of supported Cursor Icons: https://docs.rs/winit/latest/winit/window/enum.CursorIcon.html

Proposed config options 

cmdline: 
```
--cursor-icon=XYZ
```

file: 
```
config-file: "XYZ"
```

I went ahead and opened a PR for this feature request here: https://github.com/neovide/neovide/pull/2643 

**Describe alternatives you've considered**

**Additional context**

**Comment by fredizzimo** [comment]
We have been holding back a bit with this because we want it to be more generic than just supporting a single cursor. See:
* https://github.com/neovide/neovide/issues/2090

That in turn is waiting for Neovim support:
* https://github.com/neovim/neovim/issues/21458

**Comment by Jay-Madden** [comment]
@fredizzimo Totally get waiting on those things. However, those issues are more about dynamically adjusting the cursor shape based on context. This implementation would still be required for that feature as well as you would need a default fallback in the case there is no context aware shape.

**Comment by Kethku** [comment]
Unfortunately I think fredizzimo is correct here. Its a very rare ask to change the default cursor so adding an option for such a niche use case adds too much maintenance burden. It is better to await for the upstream changes and then implement your usecase using some kind of plugin or change to your theme later. Closing as won't fix but feel free to reopen if new info arrises

**Comment by falcucci** [comment]
As discussed at discord, we have agreed about bringing this as an option due to a default pattern for almost every Apple apps, so it fits to the user's preferences to decide which set of cursor will be used.

So be aware about those 'mouseshape' limitations from neovim in case you choose i-beam cursor and do not expect it to be reactive by any content elements.

A TLDR; will be added into this option and any issue related to that will be closed as default, until neovim supports it.

**Comment by Jay-Madden** [comment]
@falcucci neat, are yall wanting to use my PR https://github.com/neovide/neovide/pull/2643 or do the feature over?

**Comment by falcucci** [comment]
@Jay-Madden sorry I didn't see your PR but I've implemented the option correctly due our context.

I'll close yours in prol of #2834 but please, keep contributing :)
